### ISSUE TITLE
Screen reader on Dialog reads out twice for most things

### ISSUE THREAD

**Issue by joshualamusga1** [microsoft/fluentui#19860]
### Environment Information

- **Package version(s)**: 7.170.2 and demo site version
- **Browser and OS versions**: Edge on Windows 10 64-bit

### Describe the issue:

With NVDA, any dialog in Fabric will read out in this order upon opening the dialog: title, description, close button, title, close button, description, primary button, default button.

Ideally it should only read the title, description, and close button status once.

### Please provide a reproduction of the issue in a codepen:

https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog.  
See "Blocking Dialog" since that one has a close button.

#### Actual behavior:

With NVDA, any dialog in Fabric will read out in this order upon opening the dialog: title, description, close button, title, close button, description, primary button, default button.

#### Expected behavior:

With NVDA, any dialog in Fabric should read out in this order upon opening the dialog: title, description, close button, primary button, default button.

(I'm not sure if close button should be read first or last in this order. If so, then Narrator also announces the close button at the wrong time.)

### Documentation describing expected behavior
https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence

**Comment by gouttierre** [comment]
@joshualamusga1 - Thanks for filing this issue with us. Our team will have a look at this issue but to set expectations, we’re likely not going to get to it immediately due to the fact we’re currently focused on larger coherence work items. 

@smhigley - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @GeoffCoxMSFT

**Comment by seniroberts** [comment]
## Here's a more specific example -----> 
#19860
### The Screen Reader Announces name property for Dialog Components twice.

- This is a behavior consistent with the dialog component... 

#### Actual : 
- Screen reader announce name property for close button twice.

#### Expected:
- The screen reader should only read relevant information once for each item. This can happen when the item has a perfectly fine name/label and a redundant accessible name or label. 
- [For example, a button with a name of “Close” and an alt tag/aria tag  which says “Close”. Only one is needed in this case](https://www.w3.org/TR/wai-aria/#aria-label)

#### User  
 - If content is read twice to assistive technology, it could interfere with the comprehension of that content or simply make the application frustrating to use. 

#### Repro Steps:
1. Open URL in Edge browser.
2. Navigate on the new document control and activate it.
3. Navigate on the side navigation tab and select Add Data tab then select Local file from side pane.
4. Turn on NVDA using Ctrl + Alt + N.
5. Navigate on the page page using Tab keyboard button.
5. Observe that Screen reader announce name property for close button twice.

#### A quick fix can be applied here: 
[code pointer](https://github.com/microsoft/fluentui/commit/96dd3aa9050190d448a64889107bd3207258321b#diff-9245ad5d74be093ab7c387d53b3fed3252b03d5edaa1e00c32a5374801cd049c)

`export class DialogContentBase extends BaseComponent<IDialogContentProps, {}> {
                iconProps={{ iconName: 'Cancel' }}
                ariaLabel={closeButtonAriaLabel}
                onClick={onDismiss as any}
                title={closeButtonAriaLabel}
              />
            )}
          </div>`
